From 423b8d1b711c8b28b3474674780c22d9310425b3 Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Wed, 10 Dec 2014 10:15:55 +0800
Subject: [PATCH 1/1] usb/gadget: rename USB Tethering interface to rndis0

Modem NCM interfaces are named as usb0~4 in the system. However,
RNDIS gadget driver creates a NET device interface named usb*
by default, which is not supported by tethering overlay. So we should
rename it to rndis0 to avoid conflict.

Change-Id: Ic84bfdc21997d0e57d030072bf1d32f78eb9f162
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-10270
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
---
 drivers/usb/gadget/android.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index db2de41..8db00b9 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -683,6 +683,10 @@ rndis_function_bind_config(struct android_usb_function *f,
 	rndis_opts = container_of(rndis->f_rndis_inst,
 		struct f_rndis_opts, func_inst);
 
+	/* re-name RNDIS NET device to avoid conflict with USB modem device */
+	snprintf(rndis_opts->net->name,
+			sizeof(rndis_opts->net->name), "%s%%d", "rndis");
+
 	rndis_opts->vendor_id = rndis->vendorID;
 	rndis_opts->manufacturer = rndis->manufacturer;
 
-- 
1.9.1

